hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerStateManagerImpl.java

```diff
@@ -303,24 +317,30 @@ public void addContainer(final ContainerInfoProto containerInfo)
 
     lock.writeLock().lock();
```

Comment:
I think, if we change the `containers` map to be a concurrent hash map, then only, `reinitialize` should need a write lock, and the rest of the locks can be read locks.

Reply:
@kerneltime , thank you for reviewing this, I think this was done with write lock because of method `getContainerIDs` as it acquires read lock and read lock can be acquired by multiple threads, so we don't want dirty data while a container being added. `containers` is instance of `ContainerStateMap` and all maps used are concurrent hash maps but still dirty read is a possibility so write lock is used in addContainer. Pls let me know if I am missing something here or my understanding not correct.